src/Components/Components/src/NavigationManager.cs

```diff
@@ -61,6 +61,18 @@ public event EventHandler<NotFoundEventArgs> OnNotFound
     // The URI. Always represented an absolute URI.
     private string? _uri;
     private bool _isInitialized;
+    private const string EnableThrowNavigationException = "Microsoft.AspNetCore.Components.Endpoints.HttpNavigationManager.EnableThrowNavigationException";
+
+    /// <summary>
+    /// Gets a value indicating whether navigation exceptions should be thrown during navigation.
+    /// </summary>
+    /// <remarks>
+    /// This property is controlled by the AppContext switch
+    /// <c>Microsoft.AspNetCore.Components.Endpoints.HttpNavigationManager.EnableThrowNavigationException</c>.
+    /// When enabled, navigation operations may throw exceptions for debugging or testing purposes.
+    /// </remarks>
+    protected static bool ThrowNavigationException =>
+        AppContext.TryGetSwitch(EnableThrowNavigationException, out var switchValue) && switchValue;
```

Comment:
This should not go in `NavigationManager`

Reply:
I imagine you are putting it here because you want to "reuse" it. However, in this case its better to just duplicate the code in each of the individual classes